configure.ac

```diff
@@ -18,8 +18,8 @@ AC_SUBST([LIBSUBID_ABI_MICRO], [libsubid_abi_micro])
 AC_SUBST([LIBSUBID_ABI], [libsubid_abi])
 
 dnl Some hacks...
-test "$prefix" = "NONE" && prefix="/usr"
-test "$prefix" = "/usr" && exec_prefix=""
+test "x$prefix" = "xNONE" && prefix="/usr"
+test "X$prefix" = "X/usr" && exec_prefix=""
```

Comment:
It would be interesting to note which shells are broken.

Reply:
Actually, it is still valid for modern shells too, but for another reason.
When variable is (or contains) symbols like `!`, `-z`, `-n` etc. then interpretation could be wrong.
For example, `test "$var" = "yes"` could be expanded to `test -n = yes`. 
By prefixing with "X" (other letters works in the same way) we can make sure that variable value will not be used as an option.
See https://www.gnu.org/software/autoconf/manual/autoconf-2.72/html_node/Limitations-of-Builtins.html